exporters/otlp/test/otlp_grpc_exporter_factory_test.cc

```diff
@@ -0,0 +1,36 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+  Cripple the build environment on purpose.
+  This is to make sure that if protobuf headers
+  are included by OtlpGrpcExporterFactory,
+  even indirectly, the build will fail.
+*/
+#define PROTOBUF_VERSION 6666666
+
+#include <gtest/gtest.h>
+
+#include "opentelemetry/exporters/otlp/otlp_grpc_exporter_factory.h"
+#include "opentelemetry/exporters/otlp/otlp_grpc_exporter_options.h"
+
```

Comment:
Maybe we can add a `static_assert` to ensure we will not break the header dependency in the future?

```cpp
#ifdef GOOGLE_PROTOBUF_VERSION
static_assert(false, "protobuf should not be included here");
#endif
```